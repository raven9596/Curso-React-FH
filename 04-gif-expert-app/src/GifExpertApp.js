import React, { Fragment, useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {


  // const [categories, setCategories] = useState(['Dragon Ball']);
  const [categories, setCategories] = useState( defaultCategories );
  
  // const handleAdd = () => {

  //   setCategories([...categories, 'HunterXHunter']);
  //   setCategories(cats => [ ...cats, 'HunterXHunter']);
  // };

  return (
		<Fragment>
			<h2>GifExpertApp</h2>
			<AddCategory setCategories={setCategories} />
			<hr />

			<ol>
				{categories.map((category, i) => (
					<GifGrid key={category}  category={category} />
				))}
			</ol>
		</Fragment>
	);

};


export default GifExpertApp;