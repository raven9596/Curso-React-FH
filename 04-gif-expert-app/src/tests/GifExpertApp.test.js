import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Prueba en <GifExpertApp />', () => {
  

  test('debe mostrarse correctamente', () => {
    
    const wrapper = shallow(<GifExpertApp />);
    
    expect( wrapper ).toMatchSnapshot();
  });
  
  test('debe de mostrar una lista de categorias', () => {
    
    const categories = ['Fairy Tail', 'One Piece'];
    const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    
  });
});

