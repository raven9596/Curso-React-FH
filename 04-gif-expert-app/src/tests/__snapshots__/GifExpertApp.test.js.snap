// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Fairy Tail"
      key="Fairy Tail"
    />
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
  </ol>
</Fragment>
`;

exports[`Prueba en <GifExpertApp /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
